### ISSUE TITLE
ASP.NET .NET Framework Website (web.config) project with Sentry fails to builds on Azure build agent

### ISSUE THREAD

**Issue by edgariscoding** [getsentry/sentry-dotnet#3241]
### Package

Sentry

### .NET Flavor

.NET Framework

### .NET Version

4.7.2

### OS

Android

### SDK Version

4.2.1

### Self-Hosted Sentry Version

24.2.0

### Steps to Reproduce

We have a large project and I'm trying to integrate Sentry into it. I got it to work flawlessly on my local environment, everything builds fine and it logs to our Sentry server.

But when I push my code to Azure and run a build pipeline on it, it fails with an error:

```
App_Code\Global.asax.cs(14,0): Error CS0234: The type or namespace name 'AspNet' does not exist in the namespace 'Sentry' (are you missing an assembly reference?)
```

I installed Sentry, Sentry.AspNet, and Sentry.EntityFramework from Nuget. Like I said it all builds and works fine locally.

This type of ASP.NET website does not use a .csproj file, it instead uses only a web.config file. 

I see that we currently have a number of compilation assemblies added to the web.config. For third party tools like DevExpress, etc. Do I need to add the sentry assemblies here? Was it supposed to add these lines automatically?

If I manually add an assembly to the web.config such as this:

```
...
    <add assembly="Sentry.AspNet, Version=4.2.1.0, Culture=neutral, PublicKeyToken=fba2ec45388e2af0" />
  </assemblies>
</compilation>
```

Then instead I get this error:

```
web.config(534,0): Error ASPCONFIG: Could not load file or assembly 'Sentry.AspNet, Version=4.2.1.0, Culture=neutral, PublicKeyToken=fba2ec45388e2af0' or one of its dependencies. The system cannot find the file specified.
```

I've verified that the packages are present in the packages directory. I've verified that the bin folder contains the necessary .refresh files for the packages with the correct path to the assembly within the packages folder. 

We have plenty of other third party packages we use and have not run into this issue with any of them.

The error says that it couldn't load the assembly OR ONE OF ITS DEPENDENCIES. Could that be the problem? How do I figure out which dependency is not loading?

### Expected Result

Expected the build to succeed.

### Actual Result

Build error.

**Comment by jamescrosswell** [comment]
I'm not sure if this will help (I've never tried to deploy an ASP.NET app to Azure)... but you can [see the dependencies on NuGet.org](https://www.nuget.org/packages/Sentry.AspNet).

### Sentry.AspNet
* [Sentry](https://www.nuget.org/packages/Sentry/) (>= 4.2.1)
* [Sentry.DiagnosticSource](https://www.nuget.org/packages/Sentry.DiagnosticSource/) (>= 4.2.1)

And both of those have their own dependencies for net462.

### Sentry.DiagnosticSource 
* [Sentry](https://www.nuget.org/packages/Sentry/) (>= 4.2.1)
* [System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/) (>= 4.5.0)

### Sentry
* [System.Reflection.Metadata](https://www.nuget.org/packages/System.Reflection.Metadata/) (>= 5.0.0)
* [System.Runtime.InteropServices.RuntimeInformation](https://www.nuget.org/packages/System.Runtime.InteropServices.RuntimeInformation/) (>= 4.3.0)
* [System.Text.Json](https://www.nuget.org/packages/System.Text.Json/) (>= 6.0.8)

The error message you're getting doesn't seem to be related to Sentry though:
> Error CS0234: The type or namespace name 'AspNet' does not exist

If it was Sentry, I'd expect it to reference the 'Sentry.AspNet' namespace... as I say, I've never tried to deploy an ASP.NET app to Azure but [is ASP.NET itself installed](https://learn.microsoft.com/en-us/azure/app-service/configure-language-dotnet-framework) on the box by default? If you remove Sentry from the solution, does it deploy and run correctly?

**Comment by edgariscoding** [comment]
@jamescrosswell, so this is just a build agent. All it does is download the source code from git and build the project and it provides the build output as an artifact that we then separately deploy to our web server.

It was building fine, but it won’t succeed if I include Sentry.

I’ve integrated Sentry into a different ASP.NET Web API 2 project and it builds on azure build agents and works perfectly. The issue here is only with this ASP.NET website (with web.config).

**Comment by jamescrosswell** [comment]
> so this is just a build agent. All it does is download the source code from git and build the project and it provides the build output as an artifact that we then separately deploy to our web server.

Undertsood. My thinking was that it's a question of how the build agent differs from your local development machine (where the build works just fine). What's different buiding on the build agent vs building on your local dev machine? Is it the installation of .NET Framework? Environment variables? Possibly you're using a different web.config on thje build machine but I imagine you've tried a release build on your local dev machine and you'd know about any transformations that might be made to web.config during the build pipeline?

**Comment by edgariscoding** [comment]
> > so this is just a build agent. All it does is download the source code from git and build the project and it provides the build output as an artifact that we then separately deploy to our web server.
> 
> Undertsood. My thinking was that it's a question of how the build agent differs from your local development machine (where the build works just fine). What's different buiding on the build agent vs building on your local dev machine? Is it the installation of .NET Framework? Environment variables? Possibly you're using a different web.config on thje build machine but I imagine you've tried a release build on your local dev machine and you'd know about any transformations that might be made to web.config during the build pipeline?

I'm not sure what was causing the issue but after adding a build task that printed the contents of the bin folder after a build failure I found that some of the required dlls weren't present. 

The dll for `Sentry`, `Sentry.AspNet`, and `Sentry.EntityFramework` do not get copied to the bin folder... but `Sentry.DiagnosticSource` gets copied. 

The `.refresh` files within the bin folder have a correct path to the dlls in the packages directory. 

No idea why this wasn't working but my workaround was to add a few Bash tasks to copy the files from the packages directory to the bin directory manually.

**Comment by jamescrosswell** [comment]
Thanks for the update @edgariscoding. I'll see if I can reproduce here but it sounds like you're unblocked for the time being.

**Comment by bitsandfoxes** [comment]
Looks like you can set up a [local agent](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/windows-agent?view=azure-devops) in a Windows environment. Another thing to add to CI?

**Comment by jamescrosswell** [comment]
Closing this issue due to inactivity. If this is still relevant for you, please feel free to reopen or create a new issue with up-to-date information and we'll be happy to take another look.

Thank you for taking the time to report this!
